wilcoxomegawallfunction/wilcoxOmegaWallFunction/WilcoxomegaWallFunctionFvPatchScalarField.C

```diff
@@ -45,7 +45,7 @@ scalar WilcoxomegaWallFunctionFvPatchScalarField::tolerance_ = 1e-5;
 
 void WilcoxomegaWallFunctionFvPatchScalarField::checkType()
 {
-    if (isA<wallFvPatch>(patch()) == false)
+    if (!isA<wallFvPatch>(patch()))
```

Comment:
Do you confirm that "if !cond" is better than "if cond == false" ?
Indeed, the foamStyleCheck tool provided here (https://github.com/petebachant/foamStyleCheck) fails with this kind of style : "error T012: negation operator used in its short form"


Also, do you use a tool similar to foamStyleCheck?

Reply:
I would disagree with the foamStyleCheck in that case. There are several places where values can/should be coerced into a bool but where comparison against a bool may not make any sense.

Examples,
```
std::ostream os ....;

if (os)   // Test if stream is good
{ }

std::unique_ptr<foo> ptrA;
...
if (ptrA)   // Test if unique_ptr actually holds something
{}

foo* ptrB;
...
if (ptrB)   // Test if raw ptr actually is set
{}
```

In the last example, could reasonably argue that `if (ptrB != nullptr)` would be better style, but in any case `if (ptrB != false)` is not.

For the OpenFOAM `isA<T>` function, that was changed from returning a bool to returning a const pointer for the OpenFOAM-v1812 version. Thus you can't compare to a bool directly, but could cast to a bool and then compare:
```
if (bool(isA<foo>(obj)) == false)
```
This will work for all versions, but really does not seem like good style at all!!

The background for the isA change:
> STYLE: return a pointer instead of bool for isA<>
> - in some cases this means you could avoid an isA<> followed by
>    second dynamicCast<>

Reply:
Added as [foamStyleCheck issue](https://github.com/petebachant/foamStyleCheck/issues/3)

Reply:
I'd first note that `foamStyleCheck` is of course not official but more of a helper tool.

Is it possible to use the `not` keyword in this case, e.g.,

```c++
if (not isA<wallFvPatch>(patch()))
```

I believe that would pass.

Reply:
Hey @petebachant - good to hear from you again.
The `not` is a C++ trigraph - will be removed in c++17 so quite literally only a temporary solution.